packages/keyring/src/pair/index.ts

```diff
@@ -175,7 +175,7 @@ export function createPair ({ toSS58, type }: Setup, { publicKey, secretKey }: P
 
       const { nonce, sealed } = naclSeal(u8aToU8a(message), convertSecretKeyToCurve25519(secretKey), convertPublicKeyToCurve25519(u8aToU8a(recipientPublicKey)), nonceIn);
```

Comment:
This code, especially the `convertPublicKeyToCurve25519()` call, is likely to fail (depending on the bytes generated)

```javascript
    const sr25519Pair = keyring.createFromUri(mnemonicGenerate(), { name: 'an sr25519 pair' }, 'sr25519');
    const sr25519Curve = convertPublicKeyToCurve25519(u8aToU8a(sr25519Pair.publicKey));
    console.log(sr25519Curve);  //    <------ returns a null value
```

Reply:
My mistake. I forgot to replace ed25519 to sr25519 in testing. I have done the testing and this fix is not correct yet.